src/raft.rs

```diff
@@ -1757,6 +1803,10 @@ impl<T: Storage> Raft<T> {
     /// For a message, commit and send out heartbeat.
     pub fn handle_heartbeat(&mut self, mut m: Message) {
         self.raft_log.commit_to(m.get_commit());
+        if self.pending_request_snapshot != INVALID_INDEX {
```

Comment:
You may need to check whether `AppendResponse` set `recent_active` to true.

Reply:
`AppendResponse` does set `recent_active` to true